### PR TITLE
Write-Support for EcoTouch binding

### PR DESCRIPTION
**Pull request by sibbi77**
This PR adds write support and new tags.

To let the user specify a negative adaption value for heating (-2.0K .. +2.0K) the grammar sitemap.xtext is extended.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java [hunk 1]
```diff
@@ -124,20 +126,48 @@ protected void execute() {
...
 					if (item.getType() == EcoTouchTags.Type.Analog) {
 						// analog value encoded as a scaled integer
 						BigDecimal decimal = new BigDecimal(heatpumpValue)
 								.divide(new BigDecimal(10));
-						handleEventType(new DecimalType(decimal), item);
+						value = new DecimalType(decimal);
 					} else if (item.getType() == EcoTouchTags.Type.Word) {
 						// integer
-						handleEventType(new DecimalType(heatpumpValue), item);
+						if (item.getItemClass() == NumberItem.class)
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: although "==" seems to be valid when comparing classes i would prefer "NumberItem.class.equals(item.getItemClass())"

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java [hunk 2]
```diff
@@ -124,20 +126,48 @@ protected void execute() {
...
+								value = OnOffType.OFF;
+							else
+								value = OnOffType.ON;
+						}
 					} else {
 						// bit field
 						heatpumpValue >>= item.getBitNum();
 						heatpumpValue &= 1;
-						handleEventType(new DecimalType(heatpumpValue), item);
+						if (item.getItemClass() == NumberItem.class)
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: although "==" seems to be valid when comparing classes i would prefer "NumberItem.class.equals(item.getItemClass())"

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java [hunk 3]
```diff
@@ -163,10 +192,8 @@ private void handleEventType(org.openhab.core.types.State state,
 	/**
 	 * @{inheritDoc
 	 */
-	@Override
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: why did you remove the Override flag?

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java [hunk 4]
```diff
@@ -202,4 +229,67 @@ public void updated(Dictionary<String, ?> config)
...
+				value = connector.getValue(tag.getTagName());
+				int bitmask = 1 << tag.getBitNum();
+				if (command == OnOffType.OFF
+						|| Integer.parseInt(command.toString()) == 0) {
+					value = value & ~bitmask;
+				} else
+					value = value | bitmask;
+			} catch (Exception e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: please remove the comment and use proper logging of this Exception. At least on debug level you should provide some useful information

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java [hunk 5]
```diff
@@ -202,4 +229,67 @@ public void updated(Dictionary<String, ?> config)
...
+
+		try {
+			connector.setValue(tag.getTagName(), value);
+			// It does not make sense to check the returned value from
+			// setValue().
+			// Even if the tag is read only, one would get the newly set value
+			// back.
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: please remove the comment and use proper logging of this Exception. At least on debug level you should provide some useful information

**Comment by teichsta** [comment]
Hi @sibbi77, thanks for this contribution. Please find my review comments inline. Best, Thomas E.-E.

File: bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java (hunk 3 above)
Comment (sibbi77) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: my java compiler (not sure which version... Java8?) failed compiling this. There is no void updated(...) in the base classes to @Override

Comment (teichsta) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: could you please check if the eclipse has configured 1.5 compatibility?

Comment (sibbi77) [inline bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java]: I removed the newer JDK and reverted to 1.7. I cannot check this right now. But if you agree, that the function does not override a base class function, then the annotation is in fact not neccessary (or even wrong), correct?
